server/models/db/schema.ts

```diff
@@ -1,8 +1,225 @@
+  id: t.int().primaryKey({ autoIncrement: true }),
+  name: t.text(),
+  description: t.text(),
+  ...dateFields,
+});
+
+export const questionTags = sqliteTable(
+  'questionTags',
+  {
+    questionId: t.integer().references(() => questions.id),
+    tagId: t.integer().references(() => tags.id),
+    // questionId: t.int().references(() => questions.id),
+    // tagId: t.int().references(() => tags.id),
+  },
+  table => ({
+    pk: t.primaryKey({ columns: [table.questionId, table.tagId] }),
+  }),
+);
+
+export const votes = sqliteTable(
+  'votes',
+  {
+    id: t.int().primaryKey({ autoIncrement: true }),
+    questionId: t.int().references(() => questions.id),
+    voterId: t.int().references(() => users.id),
+    value: t.int().notNull(),
+    ...dateFields,
+  },
+  table => ({
+    uniqQuestionIdVotedBy: unique().on(table.questionId, table.voterId),
+  }),
+);
+
+/*
+  ===============================
+             RELATIONS
+  ===============================
+*/
+
+export const usersRelations = relations(users, ({ many }) => ({
+  questions: many(questions),
+  answers: many(answers),
+  questionComments: many(questionComments),
+  answerComments: many(answerComments),
+  // votes: many(votes),
+  // questionVersions: many(questionVersions),
+  // answerVersions: many(answerVersions),
+}));
+
+export const questionsRelations = relations(questions, ({ one, many }) => ({
+  askedByUser: one(users, {
+    fields: [questions.askerId],
+    references: [users.id],
+  }),
+  // votes: many(votes),
+  answers: many(answers),
+  questionComments: many(questionComments),
+  // tags: many(questionTags),
+  // versions: many(questionVersions),
+}));
+
+export const answerRelations = relations(answers, ({ one, many }) => ({
+  answererId: one(users, {
+    fields: [answers.answererId],
+    references: [users.id],
+  }),
+  questionId: one(questions, {
+    fields: [answers.questionId],
+    references: [questions.id],
+  }),
+  comments: many(answerComments),
+  // versions: many(answerVersions),
+}));
+
+export const questionsCommentsRelations = relations(questionComments, ({ one }) => ({
+  question: one(questions, {
+    fields: [questionComments.questionId],
+    references: [questions.id],
+  }),
+  commenterId: one(users, {
+    fields: [questionComments.commenterId],
+    references: [users.id],
+  }),
+}));
+
+export const answersCommentsRelations = relations(answerComments, ({ one }) => ({
+  answers: one(answers, {
+    fields: [answerComments.questionId],
+    references: [answers.id],
+  }),
+  commenterId: one(users, {
+    fields: [answerComments.commenterId],
+    references: [users.id],
+  }),
+}));
+
+// export const questionVersionsRelations = relations(questionVersions, ({ one }) => ({
+//   question: one(questions, {
+//     fields: [questionVersions.questionId],
+//     references: questions.id,
+//   }),
+//   editor: one(users, {
+//     fields: [questionVersions.editorId],
+//     references: users.id,
+//   }),
```

Comment:
references should be a list, and i think we should give a relation name because drizzle sometimes finds it hard to automatically infer

Reply:
That seems to make it work

<details>
<summary>all relations</summary>

export const usersRelations = relations(users, ({ many }) => ({
  questions: many(questions),
  answers: many(answers),
  questionComments: many(questionComments),
  answerComments: many(answerComments),
  votes: many(votes),
  questionVersions: many(questionVersions),
  answerVersions: many(answerVersions),
}));

export const questionsRelations = relations(questions, ({ one, many }) => ({
  askedByUser: one(users, {
    fields: [questions.askerId],
    references: [users.id],
  }),
  votes: many(votes),
  answers: many(answers),
  questionComments: many(questionComments),
  tags: many(questionTags),
  versions: many(questionVersions),
}));

export const answerRelations = relations(answers, ({ one, many }) => ({
  answererId: one(users, {
    fields: [answers.answererId],
    references: [users.id],
  }),
  questionId: one(questions, {
    fields: [answers.questionId],
    references: [questions.id],
  }),
  comments: many(answerComments),
  versions: many(answerVersions),
}));

export const questionsCommentsRelations = relations(questionComments, ({ one }) => ({
  question: one(questions, {
    fields: [questionComments.questionId],
    references: [questions.id],
  }),
  commenterId: one(users, {
    fields: [questionComments.commenterId],
    references: [users.id],
  }),
}));

export const answersCommentsRelations = relations(answerComments, ({ one }) => ({
  answers: one(answers, {
    fields: [answerComments.questionId],
    references: [answers.id],
  }),
  commenterId: one(users, {
    fields: [answerComments.commenterId],
    references: [users.id],
  }),
}));

export const questionVersionsRelations = relations(questionVersions, ({ one }) => ({
  question: one(questions, {
    fields: [questionVersions.questionId],
    references: [questions.id],
  }),
  editor: one(users, {
    fields: [questionVersions.editorId],
    references: [users.id],
  }),
}));

export const answerVersionsRelations = relations(answerVersions, ({ one }) => ({
  answer: one(answers, {
    fields: [answerVersions.answerId],
    references: [answers.id],
  }),
  editor: one(users, {
    fields: [answerVersions.editorId],
    references: [users.id],
  }),
}));

export const tagsRelations = relations(tags, ({ many }) => ({
  questionTags: many(questionTags),
}));

export const questionTagsRelations = relations(questionTags, ({ one }) => ({
  question: one(questions, {
    fields: [questionTags.questionId],
    references: [questions.id],
  }),
  tag: one(tags, {
    fields: [questionTags.tagId],
    references: [tags.id],
  }),
}));

export const votesRelations = relations(votes, ({ one }) => ({
  voterId: one(users, {
    fields: [votes.voterId],
    references: [users.id],
  }),
  question: one(questions, {
    fields: [votes.questionId],
    references: [questions.id],
  }),
}));


</details>

only thing is, i'm still getting a TS error here

export const votes = sqliteTable(
  'votes',
  {
    id: t.int().primaryKey({ autoIncrement: true }),
    questionId: t.int().references(() => questions.id),
    voterId: t.int().references(() => users.id),
    value: t.int().notNull(),
    ...dateFields,
  },
  table => ({
    uniqQuestionIdVotedBy: unique().on(table.questionId, table.voterId),
  }),
);